apps/cowswap-frontend/src/modules/tradeWidgetAddons/containers/MetamaskTransactionWarning/index.tsx

```diff
@@ -26,36 +28,22 @@ const NetworkInfo = styled.div`
   gap: 8px;
 `
 
-export function MetamaskTransactionWarning({ sellToken }: { sellToken: Currency }) {
-  const walletDetails = useWalletDetails()
-  const isMetamaskBrowserExtension = useIsMetamaskBrowserExtensionWallet()
-
-  const widgetProviderMetaInfo = useWidgetProviderMetaInfo()
-  const isMetamaskMobileInjectedBrowser = useIsMetamaskMobileInjectedWallet()
-
-  const isMetamaskViaWalletConnect = METAMASK_WALLET_NAME_REGEX.test(walletDetails.walletName || '')
-
-  const isWidgetMetamaskBrowserExtension = widgetProviderMetaInfo?.providerEip6963Info?.rdns === METAMASK_RDNS
-  const isWidgetMetamaskViaWalletConnect = METAMASK_WALLET_NAME_REGEX.test(
-    widgetProviderMetaInfo?.providerWcMetadata?.name || '',
-  )
+const VERSION_WHERE_BUG_WAS_FIXED = '12.10.4' // Anything smaller than this version is potentially affected by the bug
 
-  const isMetamask =
-    isMetamaskBrowserExtension ||
-    isMetamaskViaWalletConnect ||
-    isMetamaskMobileInjectedBrowser ||
-    isWidgetMetamaskBrowserExtension ||
-    isWidgetMetamaskViaWalletConnect
+export function MetamaskTransactionWarning({ sellToken }: { sellToken: Currency }) {
   const isNativeSellToken = getIsNativeToken(sellToken)
 
-  if (!isMetamask || !isNativeSellToken) return null
+  const shouldDisplayMetamaskWarning = useShouldDisplayMetamaskWarning()
+
+  if (!shouldDisplayMetamaskWarning || !isNativeSellToken) return null
 
   const chainInfo = CHAIN_INFO[sellToken.chainId as SupportedChainId]
 
   return (
     <Banner bannerType="danger" iconSize={32}>
-      Issues have been reported with Metamask sending transactions to the wrong chain. Before you sign, please check in
-      your wallet that the transaction is being sent to the network:{' '}
+      Issues have been reported with Metamask sending transactions to the wrong chain on versions prior to{' '}
```

Comment:
Nitpick: Explicitly state their **current MetaMask version** is older than v{VERSION_WHERE_BUG_WAS_FIXED}. Might be a nudge for some to upgrade?

**More aggressive:**
Or say your MetaMask version is v{YOUR_METAMASK_VERSION} and is outdated. Issues have been reported......

Reply:
I thought about it but didn't want to make the message too long.
If you think it's ok, I'll update.

Reply:
How about this 
![image](https://github.com/user-attachments/assets/c59722c7-1723-4c1a-a262-a2f1c91e9536)

If the version can be detected, the first sentence will be displayed.